core/src/main/scala/akka/persistence/cassandra/journal/CassandraJournal.scala

```diff
@@ -338,12 +373,179 @@ class CassandraJournal(cfg: Config) extends AsyncWriteJournal
     }
   }
 
+  /**
+   * Not thread safe. Assumed to only be called from the journal actor.
+   * However, unlike asyncWriteMessages it can be called before the previous Future completes
+   */
+  override def asyncDeleteMessagesTo(persistenceId: String, toSequenceNr: Long): Future[Unit] = {
+
+    // TODO could "optimize" away deletes that overlap?
+    pendingDeletes.get(persistenceId) match {
+      case null =>
+        log.debug("No outstanding delete for persistence id {}. Sequence nr: {}", persistenceId, toSequenceNr)
+        // fast path, no outstanding deletes for this persistenceId
+        val p = Promise[Unit]()
+        pendingDeletes.put(persistenceId, List(PendingDelete(persistenceId, toSequenceNr, p)))
```

Comment:
these could be mixed in any order on two concurrent calls for the same `persistenceId` leading to one getting lost for example

Reply:
Don't think they can as per previous comment